File: windows/GosaidUI/Paths.cs
using System;
using System.IO;

namespace GosaidUI;

// Mirrors internal/platform/paths.go so the UI reads/writes the same files
// the daemon does: %AppData%\gosaid\config.json and %LocalAppData%\gosaid\.
internal static class Paths
{
    private const string AppName = "gosaid";

    public static string ConfigDir =>
        Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), AppName);

    public static string ConfigFile => Path.Combine(ConfigDir, "config.json");

    public static string StateDir =>
        Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), AppName);

    public static string LogFile => Path.Combine(StateDir, "gosaid.log");

    public static string DaemonExe
    {
        get
        {
            var dir = AppContext.BaseDirectory;
            return Path.Combine(dir, "gosaid.exe");
        }
    }
}

File: macos/GosaidMenuBar/Sources/GosaidMenuBar/Paths.swift
import Foundation

// Mirrors internal/platform/paths.go.  macOS uses
// ~/Library/Application Support/gosaid for config and
// ~/Library/Logs/gosaid for logs.
enum Paths {
    private static let appName = "gosaid"

    static var configDir: URL {
        FileManager.default
            .urls(for: .applicationSupportDirectory, in: .userDomainMask)
            .first!
            .appendingPathComponent(appName)
    }

    static var configFile: URL {
        configDir.appendingPathComponent("config.json")
    }

    static var logDir: URL {
        FileManager.default
            .urls(for: .libraryDirectory, in: .userDomainMask)
            .first!
            .appendingPathComponent("Logs")
            .appendingPathComponent(appName)
    }

    static var logFile: URL {
        logDir.appendingPathComponent("gosaid.log")
    }

    /// Path to the bundled Go daemon.  When running inside the `.app`
    /// bundle this resolves to `Contents/MacOS/gosaid`.  When running a
    /// `swift run` dev build we fall back to `$PATH`.
    static var daemonExe: URL {
        let bundled = Bundle.main.bundleURL
            .appendingPathComponent("Contents/MacOS/gosaid")
        if FileManager.default.isExecutableFile(atPath: bundled.path) {
            return bundled
        }
        return URL(fileURLWithPath: "/usr/local/bin/gosaid")
    }
}
